src/main/java/se/bjurr/violations/comments/lib/model/CommentsProvider.java

```diff
@@ -4,23 +4,49 @@
 import se.bjurr.violations.lib.util.Optional;
 
 public interface CommentsProvider {
-  void createCommentWithAllSingleFileComments(String string);
 
-  void createSingleFileComment(ChangedFile file, Integer line, String comment);
-
-  List<Comment> getComments();
-
-  List<ChangedFile> getFiles();
-
-  void removeComments(List<Comment> comments);
-
-  boolean shouldComment(ChangedFile changedFile, Integer line);
-
-  boolean shouldCreateCommentWithAllSingleFileComments();
-
-  boolean shouldCreateSingleFileComment();
-
-  boolean shouldKeepOldComments();
-
-  Optional<String> findCommentTemplate();
+	/**
+	 * Create a single comment containing all file comments and violations.
+	 * @param string The combined comments.
+	 */
+	void createBulkComment(String string);
+
+	/**
+	 * Create a discussion on the diff on the line the violation occurred.
+	 *
+	 * @param file The file in which the violation occurred.
+	 * @param discussionContent The text of the discussion.
+	 * @param newLine The line number after the patch.
+	 * @param oldLine The line number before the patch.
+	 */
+	void createDiffDiscussion(ChangedFile file, String discussionContent,
```

Comment:
I think `comment` is a better word to use here. Because that is in the name of the library and all other supporting libraries and tools. Having a library named `violation-comments-lib` that creates *comments* is better then having it create *discussions*.